### ISSUE TITLE
evidence: emit events on validating new evidence

### ISSUE THREAD

**Issue by cmwaters** [tendermint/tendermint#6763]
## Summary

We emit events when evidence gets committed on chain but in adverse conditions this is not always guaranteed. Given that most severe attacks will be with greater than 1/3 of the network power, it is highly likely that the cabal will censor the evidence until it becomes expired. Until we build better mechanisms for resisting such censorship, the evidence reactor should also emit events upon successfully validating new evidence. This can better warn interested parties and can aid with social consensus. 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by jmalicevic** [comment]
This issue will be extended to add metrics on evidence validation. Should we add metrics when evidence is validated, or wait until it is committed? 

Adding a metric after evidence is validated would help discover potential misbehaviour w/o waiting for it to be committed. Then we reset the metric once it is removed from the evidence pool.

**Comment by cmwaters** [comment]
Yes I think the metric should be when it is validated. Basically it should track the size of the evidence pool
